modules/social_features/social_follow_taxonomy/modules/social_follow_tag/social_follow_tag.module

```diff
@@ -0,0 +1,206 @@
+        /** @var \Drupal\taxonomy\TermInterface $term */
+        $term = $term_storage->load($term_id);
+
+        if (!empty($term)) {
+          // Creates a link to a search page with a tag as a filter parameter.
+          $parameter = 'tag';
+          $route = 'view.search_content.page_no_value';
+          $parents = $term_storage->loadParents($term_id);
+          if (!empty($parents)) {
+            $parent = reset($parents);
+            $category = $parent->getName();
+
+            if ($tag_service = Drupal::getContainer()->get('social_tagging.tag_service')->allowSplit()) {
+              $parameter = social_tagging_to_machine_name($category);
+            }
+          }
+          $url = Url::fromRoute($route, [
+            $parameter . '[]' => $term_id,
+          ]);
+
+          // Adding additional data to the term variable to extend the template.
+          $variables['term'] = [
+            'url' => $url->toString(),
+            'name' => $term->getName(),
+            'flag' => _social_follow_taxonomy_flag_link($term),
+            'related_content' => _social_follow_tag_related_content($term),
+          ];
+        }
+      }
+      break;
+  }
+}
+
+/**
+ * Extends variables for social_tagging_split template.
+ *
+ * Implements hook_preprocess_HOOK().
+ * {@inheritdoc}
+ */
+function social_follow_tag_preprocess_social_tagging_split(&$variables) {
+  // Iterate over an array with parent terms (category).
+  foreach ($variables['taghierarchy'] as $parent_tag) {
+    // Iterate over an array with terms in each category.
+    foreach ($parent_tag['tags'] as $tag_id => $tag_info) {
+      /** @var \Drupal\taxonomy\TermStorageInterface $term_storage */
+      $term_storage = \Drupal::entityTypeManager()->getStorage('taxonomy_term');
+      $current_term = $term_storage->load($tag_id);
+      // Must be a valid Term.
+      if (!$current_term instanceof TermInterface) {
+        continue;
+      }
+      // Get current term parent.
+      $parents = $term_storage->loadParents($current_term->id());
+      if (!empty($parents)) {
+        $parent = reset($parents);
+
+        // Adding additional data to the term variable to extend the template.
+        $variables['taghierarchy'][$parent->id()]['tags'][$current_term->id()] = [
+          'url' => $tag_info['url'],
+          'name' => $tag_info['name'],
+          'flag' => _social_follow_taxonomy_flag_link($current_term),
+          'related_content' => _social_follow_tag_related_content($current_term),
+          'follow' => _social_follow_tag_follow($current_term),
+        ];
+      }
+    }
+  }
+}
+
+/**
+ * Extends variables for social_tagging_nosplit template.
+ *
+ * Implements hook_preprocess_HOOK().
+ * {@inheritdoc}
+ */
+function social_follow_tag_preprocess_social_tagging_nosplit(&$variables) {
+  $term_storage = \Drupal::entityTypeManager()->getStorage('taxonomy_term');
+
+  foreach ($variables['tags'] as $tag_name => $tag_info) {
```

Comment:
@VolodymyrMartsynenko  Are we sure that an array always will be passed to `$variables['tags']`?
If not - maybe good to add some additional check here.

Reply:
I added an additional check for this.